### PR TITLE
Use random name for manager semaphore

### LINKED ISSUE #56: Semaphore name needs to be randomized, otherwise clash with other users is unavoidable
On some of our cluster nodes, when I run Ignition Blueprint/ign-launch1 (the SubT simulator), I get an error:

    [Err] [Manager.cc:300] Error initializing semaphore: Permission denied

That points to

https://github.com/ignitionrobotics/ign-launch/blob/9f4a9298b2f549cd0df66d30f9c5942198593ea5/src/Manager.cc#L298

which can fail in case the semaphore with name 

https://github.com/ignitionrobotics/ign-launch/blob/9f4a9298b2f549cd0df66d30f9c5942198593ea5/src/Manager.cc#L50

already exists and is created by another user with permissions that don't allow modification of the semaphore (i.e. 644).

I've checked that on the failing nodes, the semaphore indeed exists and is created by someone else:

    $ ls -al /dev/shm/sem.*
    -rw-r--r-- 1 otheruser rob 32 Aug 17 22:36 /dev/shm/sem.child_semaphore

In such a case, I can't use Ignition launch at all on that cluster node.

It'd be much better to name the semaphore either randomly, or predictably incorporating the username. Some other apps create semaphores named e.g.:

    -rw-------. 1 user group 32 Aug 27 17:42 /dev/shm/sem.loky-447964-ctehrdwd
    -rw-------. 1 user group 32 Aug 27 17:42 /dev/shm/sem.loky-447964-dfqrdehz
    -rw-------. 1 user group 32 Aug 27 17:42 /dev/shm/sem.loky-447964-ed6_sluy
    -rw-------. 1 user group 32 Nov 20  2019 /dev/shm/sem.mp-29fggu5g
    -rw-------. 1 user group 32 Dec 16  2019 /dev/shm/sem.mp-2croga40

### PR DESCRIPTION
**Pull request by peci1**
Fixes #56.

### CODE REVIEW

**Comment by chapulina** [comment]
Please be sure to sign your commit for the DCO checker.

**Comment by peci1** [comment]
DCO fixed.

File: src/Manager.cc [hunk 1]
```diff
@@ -295,9 +297,11 @@ ManagerPrivate::ManagerPrivate()
       std::bind(&ManagerPrivate::OnSigIntTerm, this, std::placeholders::_1));
 
   // Initialize semaphore
-  this->stoppedChildSem = sem_open(kSemaphoreName, O_CREAT, 0644, 1);
+  this->stoppedChildSemName = std::string("ign-launch-") + common::Uuid().String();
+  this->stoppedChildSem = sem_open(this->stoppedChildSemName.c_str(), O_CREAT, 0644, 1);
```
Comment (mjcarroll) [inline src/Manager.cc:301]: This causes issues in MacOS CI, as the new name is too long.  I'm not sure exactly what `SEM_NAME_LEN` is on MacOS, but a good guess would be 31 from something like: https://chromium.googlesource.com/native_client/src/native_client/+/d6d53e83fffcc64b6e0d52acc31c63e0b5a60186/src/shared/platform/osx/nacl_semaphore.c#52

Comment (mjcarroll) [reply src/Manager.cc:301]: Maybe something like `mktemp` would be more suitable here.

Comment (peci1) [reply src/Manager.cc:316]: Okay, I posted an alternative solution which should generate about 22-char filenames.

I haven't found a nice way to strongly seed the Rand() generator, so the seeding part is a bit ugly. `mktemp` doesn't sound good to me - it still seems to me that this pseudorandom magic would be faster than creating a file just for reading its name and closing it.

Comment (peci1) [reply src/Manager.cc:316]: I needed to add a private dependency of libignition-launch1.so on libignition-math6.so. I hope it's not considered as ABI break.

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**